src/v/cluster/partition.cc

```diff
@@ -84,6 +85,23 @@ partition::partition(
             }
         }
     }
+
+    _log_cleanup_policy.watch([this]() {
+        if (_as.abort_requested()) {
+            return ss::now();
+        }
+        auto changed = _raft->log()->notify_compaction_update();
```

Comment:
Can the log not manage this itself, with the config passed through log_manager?

Reply:
I could have a watcher in `log_manager` that calls `log->notify_compaction_update()`, but (1) we would still need `partition` to know about changes to the compaction status to ensure that `archiver` is restarted and (2) it seems that `partition` already manages the compaction status of the log through `partition::update_configuration`.